Mlem/App/Utility/Extensions/Content Models/Comment1Providing+Extensions.swift

```diff
@@ -188,4 +192,34 @@ extension Comment1Providing {
             callback: callback
         )
     }
+    
+    func collapseAction(commentTreeTracker: CommentTreeTracker? = nil) -> BasicAction {
+        .init(
+            id: "collapse\(uid)",
+            appearance: .collapse(),
+            callback: { @MainActor in
+                commentTreeTracker?.nodesKeyedByActorId[self.actorId]?.collapsed = true
```

Comment:
We can use `withAnimation` here to give swipe-to-collapse the same animation as when you tap-to-collapse:

```suggestion
                withAnimation(UIAccessibility.isReduceMotionEnabled ? nil : .default) {
                    commentTreeTracker?.nodesKeyedByActorId[self.actorId]?.collapsed = true
                }
```